### DISCUSSION TITLE
Can't get EMAIL_DOMAIN_ALLOWLIST to stop other domains signing up

### DISCUSSION THREAD

**Discussion by metakarma** [mastodon/mastodon#17830] [Q&A] [answered]
Hi,

First Mastodon install. Loving the ease so far. Here is my very first glitch.

I would like to have only a few domains be able to use this instance - say anyone with metakarma.org or metakarma.net emails.

So in my ./live directory, I edited .env.production to add:
EMAIL_DOMAIN_ALLOWLIST=metakarma.org|metakarma.net

I restarted the server with:
systemctl restart mastodon-web
systemctl restart mastodon-streaming
systemctl restart mastodon-sidekiq

I go to my instance and sign up a new account with a gmail.com address ... and it works!

Any suggestions how I can get this to work?

Many thanks, Tony

**Comment by metakarma** [comment, marked as answer]
Just answering my own question - this has got to do with Mastodon versions.

I have been using the Digital Ocean droplet image for my install. This creates a 3.1.3 version, not the latest 3.4.6

Variables have been renamed between versions. 

WHITELIST_MODE=true
EMAIL_DOMAIN_WHITELIST=[...]

work on 3.1.2

**Reply by ClearlyClaire** [reply]
3.1.2 is seriously outdated and you should update, many security issues have been discovered and fixed since that version

**Reply by metakarma** [reply]
ok ... thanks!
